### DISCUSSION TITLE
Environment variables listed in a file.

### DISCUSSION THREAD

**Discussion by AHuusom** [actions/virtual-environments#3752] [Ideas]
As we are trying to create ubuntu build servers as automatically as possible, we ran into a problem. As the Azure Devops agent uses the .env and the .path files for the environment variables. And these two files are created by running a env.sh file in the agent folder. This env.sh has som standard Environment Variable keys defined which it then copies into the .env file and if you want all of the environment variables you need to update this env.sh file with all the EV keys.
My question is then: Is it possible that all the environment variables keys created during the image creation could be saved in a txt file or equivalent in the post-generation folder for example.
Then we can modify the env.sh script to look in this file as we have done today and this works.
But we need to manually update the file in the post-generation folder with new environment variables.

**Comment by miketimofeev** [comment]
Hi @AHuusom! All the env variables are already saved in `/etc/environment` file but one important moment — you need to run this script on the deployed VM once to replace variable related to user's home directory with proper values https://github.com/actions/virtual-environments/blob/main/images/linux/post-generation/environment-variables.sh
